### PR TITLE
Report status via API

### PR DESCRIPTION
**Pull request by asomov**
This request accumulated a lot of changes.
1) Remove blocking Await.result() call from EventLoggingActor
2) Keep the status inside database actors (the error messages are not kept, only the timestamps of errors to be found in logs)
3) Introduce StatusReporterActor to get the statuses
4) Write a unit test for StatusReporterActor (as example to write tests for actors)
5) Add diagram for StatusReporterActor to documentation
6) Expose status via "/status" endpoint, expose the build date and the current server date
7) Introduce DateUtils to work with date formats in the same way
8) Minor changes in Event to be able to receive events from meteor

### CODE REVIEW

**Comment by wouterd** [comment]
Hi @asomov I see some merge conflicts that need to be resolved. LGTM otherwise.
Not sure about how you convert the millis to a string representation of the date. Seems like the build info from the build server is now possibly interpreted in another timezone. Is it safe to assume that all times are in UTC (Zulu time)?

**Comment by asomov** [comment]
Merged.
Yes, the time must be UTC.

File: src/main/scala/com/partup/EventLoggingActor.scala [hunk 1]
```diff
@@ -1,33 +1,69 @@
...
 class EventLoggingActor(mongo: MongoClient) extends Actor with ActorLogging {
+
+  var status = EventLoggingStatus(0, 0, 0, Nil, DateUtils.now)
+
   override def receive = {
     case e: RawEvent =>
       val doc = Document(e.toJson.compactPrint)
 
-      val obs = mongo
+      val obs: Observable[Completed] = mongo
```
Comment (wouterd) [inline src/main/scala/com/partup/EventLoggingActor.scala]: No need to type this explicitly. Also not part of the scope of this PR. 
Try to keep opinionated code quality things in their own PRs.

File: src/main/scala/com/partup/EventRoutingActor.scala [hunk 2]
```diff
@@ -81,7 +81,8 @@ class EventRoutingActor(neo4jactor: ActorRef) extends Actor with ActorLogging {
         //Teams
         case "partups.inserted" | "partups.updated" | "partups.changed" =>
           val payload = event.payload.asJsObject.fields
-          val user_id = payload("0").convertTo[String]
+          //TODO key not found: 0
```
Comment (wouterd) [inline src/main/scala/com/partup/EventRoutingActor.scala]: Leave this broken, so it's easier for Maurits to spot the problems. Now Neo4J is sent wrong data.
Please don't make functional changes in the code when doing a "refactor".

File: src/main/scala/com/partup/EventRoutingActor.scala [hunk 3]
```diff
@@ -242,8 +243,10 @@ class EventRoutingActor(neo4jactor: ActorRef) extends Actor with ActorLogging {
         //Ratings
         case "partups.contributions.ratings.inserted" =>
           val payload = event.payload.asJsObject.fields
-          val user = payload("0").asJsObject.fields
-          val _id = user("id").convertTo[String]
+          val _id: String = payload.get("0") match {
```
Comment (wouterd) [inline src/main/scala/com/partup/EventRoutingActor.scala]: Same as before, keep this broken so we know it's broken and it doesn't stealthily change Neo4j with wrong data.

File: src/main/scala/com/partup/Boot.scala [hunk 4]
```diff
@@ -18,9 +18,10 @@ object Boot extends App {
   implicit val system = ActorSystem("partup-api")
 
   val neo4j = createNeo4jActor
-  val eventReceivers = Seq(createEventLoggingActor, createEventRoutingActor(neo4j))
+  val eventLogger = createEventLoggingActor
+  val eventReceivers = Seq(eventLogger, createEventRoutingActor(neo4j))
 
-  val service = system.actorOf(EventsApiActor.props(eventReceivers))
+  val service = system.actorOf(EventsApiActor.props(eventReceivers, neo4j, eventLogger), "serviceActor")
```
Comment (wouterd) [inline src/main/scala/com/partup/Boot.scala]: Can change the parameters of EventsApiActor to (eventRouter, neo4j, eventLogger), now the loggingActor is sent over twice.

File: src/main/scala/com/partup/utils/DateUtils.scala [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    * @return ISO date
+    */
+  def now: String = OffsetDateTime.now().format(DateTimeFormatter.ISO_OFFSET_DATE_TIME)
+
+  /**
+    * ISO formatted timestamp
+    * @param timestamp - epoch timestamp to convert
+    * @return ISO date
+    */
+  def format(timestamp: Long): String = OffsetDateTime.ofInstant(Instant.ofEpochMilli(timestamp), ZoneId.systemDefault()).format(DateTimeFormatter.ISO_OFFSET_DATE_TIME)
```
Comment (wouterd) [inline src/main/scala/com/partup/utils/DateUtils.scala]: `ZoneId.systemDefault` should be the ZoneId of UTC.

**Comment by wouterd** [comment]
Check the comments, needs some fixes before we can merge.

File: src/main/scala/com/partup/EventRoutingActor.scala (hunk 2 above)
Comment (asomov) [inline src/main/scala/com/partup/EventRoutingActor.scala]: Without this change the actor crashes with every message.

File: src/main/scala/com/partup/EventRoutingActor.scala (hunk 3 above)
Comment (asomov) [inline src/main/scala/com/partup/EventRoutingActor.scala]: Added a TODO

File: src/main/scala/com/partup/EventLoggingActor.scala (hunk 1 above)
Comment (asomov) [inline src/main/scala/com/partup/EventLoggingActor.scala]: We do need to show the type for method calls. (to avoid code encryption)
This PR started with this change later it was extended to show errors instead of logging them.

File: src/main/scala/com/partup/Boot.scala (hunk 4 above)
Comment (asomov) [inline src/main/scala/com/partup/Boot.scala]: Fixed.

File: src/main/scala/com/partup/EventLoggingActor.scala (hunk 1 above)
Comment (wouterd) [inline src/main/scala/com/partup/EventLoggingActor.scala]: What's code encryption?

File: src/main/scala/com/partup/EventRoutingActor.scala (hunk 2 above)
Comment (wouterd) [inline src/main/scala/com/partup/EventRoutingActor.scala]: Yes, that's fine because we need to fix this anyway.

Comment (vanderGoes) [inline src/main/scala/com/partup/EventRoutingActor.scala]: How is it crashing?

Comment (asomov) [inline src/main/scala/com/partup/EventRoutingActor.scala]: When  the "0" key is missing the actor crashes (and re-starts)

Comment (vanderGoes) [inline src/main/scala/com/partup/EventRoutingActor.scala]: OK, again. I'll check it. Thanks
